sdk/core/Azure.Core/src/AsyncCollection.cs

```diff
@@ -67,7 +67,7 @@ public abstract class AsyncCollection<T> : IAsyncEnumerable<Response<T>> where T
         /// enumerating asynchronously.
         /// </param>
         /// <returns>An async sequence of values.</returns>
-        public virtual async IAsyncEnumerator<Response<T>> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        public virtual async IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = default)
```

Comment:
Do you also need to change the default implementation below to just `yield return value`?  I think the implicit cast is making it compile, but there's no need for the extra allocation.

Reply:
Done.